File: Questão-4/Animal.go
package main

import (
	"fmt"
)

type Animal struct {
	nome    string
	especie string
}
type Cachorro struct {
	Animal
}
type Gato struct {
	Animal
}

func (a Animal) EmitirSom() {
}
func (c Cachorro) EmitirSom() {
	fmt.Println("AuAu")
}
func (g Gato) EmitirSom() {
	fmt.Printf("Miau")
}

func main() {
	cachorro := new(Cachorro)
	cachorro.nome = "Cão"
	cachorro.especie = "Caninus Felizis"

	gato := new(Gato)
	gato.nome = "Gatinho"
	gato.especie = "Gatitus Felizis"

	fmt.Printf("Nome do Animal: %s\nEspécie do Animal: %s\nSom do Animal: ", gato.nome, gato.especie)
	gato.EmitirSom()
	fmt.Printf("\n\nNome do Animal: %s\nEspécie do Animal: %s\nSom do Animal: ", cachorro.nome, cachorro.especie)
	cachorro.EmitirSom()

}

File: Questão-4/Animal.py
class Animal:
  def __init__(self, nome, especie):
    self.nome = nome
    self.especie = especie

  def EmitirSom(self):
    pass

class Cachorro(Animal):
  def EmitirSom(self):
    print("AuAu")

class Gato(Animal):
  def EmitirSom(self):
    print("Miau")

cachorro = Cachorro("Cão", "Canis familiaris")
gato = Gato("Gatinho", "Felis catus")

print(f"\nO Animal {cachorro.nome} da espécie {cachorro.especie} faz:")
cachorro.EmitirSom()
print(f"\nO Animal {gato.nome} da espécie {gato.especie} faz:")
gato.EmitirSom()
